AutoSteamApp/Automaton/SteamworkAutomaton.cs

```diff
@@ -0,0 +1,348 @@
+﻿using AutoSteamApp.Configuration;
+using AutoSteamApp.Helpers;
+using AutoSteamApp.ProcessMemory;
+using GregsStack.InputSimulatorStandard;
+using GregsStack.InputSimulatorStandard.Native;
+using Logging;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Windows.Forms;
+
+namespace AutoSteamApp.Automaton
+{
+
+    public class SteamworkAutomaton
+    {
+
+        #region Fields
+
+        /// <summary>
+        /// Process loaded in the constructor.
+        /// </summary>
+        Process _Process;
+
+        /// <summary>
+        /// Save data loaded in the constructor.
+        /// </summary>
+        SaveData _SaveData;
+
+        /// <summary>
+        /// Steamworks data loaded in constructor.
+        /// </summary>
+        SteamworksData _SteamworksData;
+
+        /// <summary>
+        /// Field used to flag whether or not the currently running MHW:IB version is supported.
+        /// </summary>
+        bool _SupportedVersion = false;
+
+        /// <summary>
+        /// The input simulator used to mock button presses.
+        /// </summary>
+        InputSimulator _InputSimulator;
+
+        #endregion
+
+        #region Constructor
+
+        /// <summary>
+        /// Constructor which loads all required process memory and configuration values.
+        /// </summary>
+        public SteamworkAutomaton()
+        {
+            try
+            {
+                LoadAndVerifyProcess();
+                if (_SupportedVersion && !ConfigurationReader.RandomRun)
+                {
+                    _SteamworksData = new SteamworksData(_Process);
+                    _SaveData = new SaveData(_Process);
```

Comment:
An error thrown by eighter `_SaveData` or `_SteamworksData` will break the Random functionality also - as the _InputSimulator object will not have an instance eighter.

Reply:
I have wrapped the constructor in a try/catch will will exit the application if an error occurs. The random run will not be able to execute because the application will exit.

Are you suggesting that we make it do a random run instead of exiting?